ocrd_utils/ocrd_utils/os.py

```diff
@@ -128,25 +128,36 @@ def resolve_mets_arguments(directory, mets_url, mets_basename, log=None):
     """
     if mets_basename and mets_url:
         raise ValueError("Use either --mets or --mets-basename, not both")
-    if mets_basename and not mets_url:
-        (log.warning if log else print)(DeprecationWarning("--mets-basename is deprecated. Use --mets/--directory instead"))
-    if mets_url and not mets_basename:
-        mets_basename = basename(mets_url)
-    elif not mets_basename:
+    elif not mets_basename and mets_url:
+        mets_basename = Path(mets_url).name
+    elif not mets_basename and not mets_url:
         mets_basename = 'mets.xml'
+    else:
+        (log.warning if log else print)(DeprecationWarning("--mets-basename is deprecated. Use --mets/--directory instead"))
+
     if directory and mets_url:
-        directory = abspath(directory)
-        if not abspath(mets_url).startswith(directory):
-            raise ValueError("--mets has a directory part inconsistent with --directory")
+        # XXX check whether mets_url has no parents, i.e. is actually the mets_basename
+        if Path(mets_url).parent == Path('.'):
```

Comment:
That's a strong assumption though. The user might run with `--mets data/mets.xml -w data`. The old solution was simpler AFAICS (see above).

Reply:
`--mets data/mets.xml`: Then `Path(mets_url).parent` would be `Path('data')`, not `Path(.)`

Reply:
Yes, got it.